### PR TITLE
OAK-10464: Use Testcontainers instead of com.arakelian:docker-junit-rule

### PR DESCRIPTION
**Pull request by t-rana**

### CODE REVIEW

File: oak-blob-cloud-azure/pom.xml [hunk 1]
```diff
@@ -191,10 +191,10 @@
             <artifactId>mockito-core</artifactId>
             <scope>test</scope>
         </dependency>
-
         <dependency>
-            <groupId>com.arakelian</groupId>
-            <artifactId>docker-junit-rule</artifactId>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>testcontainers</artifactId>
+            <version>${testcontainers.version}</version>
```
Comment (reschke) [inline oak-blob-cloud-azure/pom.xml]: Why do we need the version property?

File: oak-blob-cloud-azure/src/test/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzuriteDockerRule.java [hunk 2]
```diff
@@ -16,55 +16,88 @@
...
-            .addStartedListener(container -> {
+    @Override
+    protected void after() {
+        if (azuriteContainer != null) {
+            azuriteContainer.stop();
+        }
+    }
+
+    @Override
+    public Statement apply(Statement base, Description description) {
```
Comment (reschke) [inline oak-blob-cloud-azure/src/test/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzuriteDockerRule.java:71]: Please avoid moving around methods when not needed.

File: oak-lucene/pom.xml [hunk 3]
```diff
@@ -471,5 +471,11 @@
       <version>1.19.0</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (reschke) [inline oak-lucene/pom.xml]: It seems we're adding unneeded dependencies here...

File: oak-segment-azure/src/test/resources/logback-test.xml [hunk 4]
```diff
@@ -21,11 +21,19 @@
             <pattern>%d{yyyy-MM-dd HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
         </encoder>
     </appender>
-    <statusListener class="ch.qos.logback.core.status.NopStatusListener" />
+    <statusListener class="ch.qos.logback.core.status.NopStatusListener"/>
```
Comment (reschke) [inline oak-segment-azure/src/test/resources/logback-test.xml]: Is this related to switching to testcontainers?

Comment (t-rana) [reply oak-segment-azure/src/test/resources/logback-test.xml]: logs were getting printed on console, it is always better to have a log file, hence added the same.

Comment (reschke) [reply oak-segment-azure/src/test/resources/logback-test.xml]: Then by all means open a separate ticket and create a second PR.

A single PR should address a single problem only (if possible).

File: oak-lucene/pom.xml (hunk 3 above)
Comment (t-rana) [reply oak-lucene/pom.xml]: oak lucene uses the MongoConnectionFactory in IT(s)  which in turns uses MongoDockerRule, test container dependency has to be added in oak lucene

https://github.com/apache/jackrabbit-oak/blob/trunk/oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/directory/ActiveDeletedBlobCollectionIT.java

Comment (reschke) [reply oak-lucene/pom.xml]: But isn't that automatically inherited?

File: oak-segment-azure/src/test/resources/logback-test.xml (hunk 4 above)
Comment (t-rana) [reply oak-segment-azure/src/test/resources/logback-test.xml]: @reschke I cannot create a ticket in Oak, if you can create one please do that, I will raise a PR for same under that ticket.

File: oak-blob-cloud-azure/pom.xml (hunk 1 above)
Comment (t-rana) [reply oak-blob-cloud-azure/pom.xml]: defining a version is always beneficial, test containers is widely used and it might pick up some different version from some other modules which might lead to version mismatch. In this case it will always refer oak-parent for test container version

File: oak-lucene/pom.xml (hunk 3 above)
Comment (t-rana) [reply oak-lucene/pom.xml]: No, testcontainers dependency is defined under dependencyManagment of oak-parent. This will not be inherited by the child module until it is explicitly defined under dependencies of child module.

Other workaround for this is to define it under the dependencies tag in parent pom, in this way it will be available to all the child modules but this should not be done since this is not needed by all the child modules

Comment (reschke) [reply oak-lucene/pom.xml]: I was/am confused because I would have assumed that switching from one docker framework to another one will *replace* a dependency (not add one)...

File: oak-segment-azure/src/test/resources/logback-test.xml (hunk 4 above)
Comment (reschke) [reply oak-segment-azure/src/test/resources/logback-test.xml]: -> https://selfserve.apache.org/jira-account.html

File: oak-lucene/pom.xml (hunk 3 above)
Comment (t-rana) [reply oak-lucene/pom.xml]: My bad, I added the new dependency of testcontainers but did not remove the spotify docker client dependency. Have  fixed the same.

File: oak-segment-azure/src/test/resources/logback-test.xml (hunk 4 above)
Comment (t-rana) [reply oak-segment-azure/src/test/resources/logback-test.xml]: I have raised a request for jira access on the portal that you shared. I guess this will take some time, can we keep this change in this PR? Will keep this  in mind in future PR(s).

File: oak-blob-cloud-azure/pom.xml (hunk 1 above)
Comment (reschke) [reply oak-blob-cloud-azure/pom.xml]: If that was true, why are we doing it differently for approximately all other dependencies?

File: oak-segment-azure/src/test/resources/logback-test.xml (hunk 4 above)
Comment (reschke) [reply oak-segment-azure/src/test/resources/logback-test.xml]: No, please move this into a separate PR.

Comment (t-rana) [reply oak-segment-azure/src/test/resources/logback-test.xml]: this change is reverted now

File: oak-blob-cloud-azure/pom.xml (hunk 1 above)
Comment (t-rana) [reply oak-blob-cloud-azure/pom.xml]: Defining a version in multi module project is always beneficial. The version which is mentioned in the child's pom always received precedence over any dependency conflicts, in this case the version is defined in parent pom which in turn is used by all child modules (this makes sure that the same version of test containers is used in oak).

I don't know why the same has not been done in other cases.

**Comment by smiroslav** [comment]
Got an error running the test in oak-segment-azure
```
> mvn clean install -Dtest=AzureRepositoryLockTest
...
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.jackrabbit.oak.segment.azure.AzureRepositoryLockTest
[ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 8.893 s <<< FAILURE! - in org.apache.jackrabbit.oak.segment.azure.AzureRepositoryLockTest
[ERROR] testLeaseRefreshUnsuccessful(org.apache.jackrabbit.oak.segment.azure.AzureRepositoryLockTest)  Time elapsed: 0.545 s  <<< ERROR!
org.mockito.exceptions.base.MockitoException:

Mockito cannot mock this class: class com.microsoft.azure.storage.blob.CloudBlockBlob.
Can not mock final classes with the following settings :
 - explicit serialization (e.g. withSettings().serializable())
 - extra interfaces (e.g. withSettings().extraInterfaces(...))

You are seeing this disclaimer because Mockito is configured to create inlined mocks.
You can learn about inline mocks and their limitations under item #39 of the Mockito class javadoc.

Underlying exception : org.mockito.exceptions.base.MockitoException: Could not modify all classes [interface com.microsoft.azure.storage.blob.ListBlobItem, class com.microsoft.azure.storage.blob.CloudBlob, class java.lang.Object, class com.microsoft.azure.storage.blob.CloudBlockBlob]
	at org.apache.jackrabbit.oak.segment.azure.AzureRepositoryLockTest.testLeaseRefreshUnsuccessful(AzureRepositoryLockTest.java:94)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.apache.jackrabbit.oak.blob.cloud.azure.blobstorage.AzuriteDockerRule$1.evaluate(AzuriteDockerRule.java:84)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: org.mockito.exceptions.base.MockitoException: Could not modify all classes [interface com.microsoft.azure.storage.blob.ListBlobItem, class com.microsoft.azure.storage.blob.CloudBlob, class java.lang.Object, class com.microsoft.azure.storage.blob.CloudBlockBlob]
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:168)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:399)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:190)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:410)
	... 30 more
Caused by: java.lang.IllegalStateException:
Byte Buddy could not instrument all classes within the mock's type hierarchy

This problem should never occur for javac-compiled classes. This problem has been observed for classes that are:
 - Compiled by older versions of scalac
 - Classes that are part of the Android distribution
	at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.triggerRetransformation(InlineBytecodeGenerator.java:284)
	at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.mockClass(InlineBytecodeGenerator.java:217)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.lambda$mockClass$0(TypeCachingBytecodeGenerator.java:47)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:168)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:399)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:190)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:410)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.mockClass(TypeCachingBytecodeGenerator.java:40)
	at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.createMockType(InlineDelegateByteBuddyMockMaker.java:396)
	at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.doCreateMock(InlineDelegateByteBuddyMockMaker.java:355)
	at org.mockito.internal.creation.bytebuddy.InlineDelegateByteBuddyMockMaker.createSpy(InlineDelegateByteBuddyMockMaker.java:345)
	at org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker.createSpy(InlineByteBuddyMockMaker.java:62)
	at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:91)
	at org.mockito.internal.MockitoCore.mock(MockitoCore.java:88)
	at org.mockito.Mockito.spy(Mockito.java:2120)
	... 30 more
Caused by: java.lang.IllegalArgumentException: Java 21 (65) is not supported by the current version of Byte Buddy which officially supports Java 20 (64) - update Byte Buddy or set net.bytebuddy.experimental as a VM property
	at net.bytebuddy.utility.OpenedClassReader.of(OpenedClassReader.java:96)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:4011)
	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:2224)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase$UsingTypeWriter.make(DynamicType.java:4050)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3734)
	at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.transform(InlineBytecodeGenerator.java:401)
	at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:244)
	at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:610)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:225)
	at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.triggerRetransformation(InlineBytecodeGenerator.java:280)
	... 44 more

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   AzureRepositoryLockTest.testLeaseRefreshUnsuccessful:94 Mockito
Mockito canno...
```

File: oak-blob-cloud-azure/pom.xml (hunk 1 above)
Comment (reschke) [reply oak-blob-cloud-azure/pom.xml]: I don't get that. We define the dependency with version in the parent POM, and then use it version-less in all subprojects. That's IMHO the right approach, and no variable is needed at all.

**Comment by t-rana** [comment]
@smiroslav , its working fine on my system, I ran the same command and it executed successfully.

% mvn clean install -Dtest=AzureRepositoryLockTest
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------< org.apache.jackrabbit:oak-segment-azure >---------------
[INFO] Building Oak Segment Azure 1.59-SNAPSHOT
[INFO] -------------------------------[ bundle ]-------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.2.0:clean (default-clean) @ oak-segment-azure ---
[INFO] Deleting /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target
[INFO] 
[INFO] --- maven-enforcer-plugin:3.1.0:enforce (enforce-maven-version) @ oak-segment-azure ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.1.0:enforce (enforce-java-version) @ oak-segment-azure ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.8:prepare-agent (pre-unit-test) @ oak-segment-azure ---
[INFO] Skipping JaCoCo execution because property jacoco.skip is set.
[INFO] test.opts.coverage set to empty
[INFO] 
[INFO] --- maven-remote-resources-plugin:1.7.0:process (process-resource-bundles) @ oak-segment-azure ---
[INFO] Preparing remote bundle org.apache:apache-jar-resource-bundle:1.4
[INFO] Copying 3 resources from 1 bundle.
[INFO] 
[INFO] --- maven-resources-plugin:3.3.0:resources (default-resources) @ oak-segment-azure ---
[INFO] skip non existing resourceDirectory /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/src/main/resources
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.10.1:compile (default-compile) @ oak-segment-azure ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 21 source files to /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/classes
[INFO] /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/tool/AzureCompact.java: /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/tool/AzureCompact.java uses or overrides a deprecated API.
[INFO] /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/tool/AzureCompact.java: Recompile with -Xlint:deprecation for details.
[INFO] 
[INFO] --- maven-bundle-plugin:5.1.8:manifest (scr-metadata) @ oak-segment-azure ---
[INFO] No MANIFEST.MF file found, generating manifest.
[WARNING] Manifest org.apache.jackrabbit:oak-segment-azure:bundle:1.59-SNAPSHOT : Export com.microsoft.azure.storage,  has 4,  private references [com.microsoft.azure.storage.analytics, com.microsoft.azure.storage.file, com.microsoft.azure.storage.queue, com.microsoft.azure.storage.table]
[WARNING] Manifest org.apache.jackrabbit:oak-segment-azure:bundle:1.59-SNAPSHOT : Export com.microsoft.azure.storage.core,  has 2,  private references [com.microsoft.azure.storage.queue, com.microsoft.azure.storage.table]
[WARNING] Manifest org.apache.jackrabbit:oak-segment-azure:bundle:1.59-SNAPSHOT : Export com.microsoft.azure.storage.blob,  has 2,  private references [com.microsoft.azure.keyvault.core, com.microsoft.azure.storage.file]
[INFO] Writing manifest: /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/classes/META-INF/MANIFEST.MF
[INFO] 
[INFO] --- maven-resources-plugin:3.3.0:testResources (default-testResources) @ oak-segment-azure ---
[INFO] Copying 2 resources
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.10.1:testCompile (default-testCompile) @ oak-segment-azure ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 27 source files to /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/test-classes
[INFO] /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/src/test/java/org/apache/jackrabbit/oak/segment/azure/fixture/SegmentAzureFixture.java: /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/src/test/java/org/apache/jackrabbit/oak/segment/azure/fixture/SegmentAzureFixture.java uses or overrides a deprecated API.
[INFO] /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/src/test/java/org/apache/jackrabbit/oak/segment/azure/fixture/SegmentAzureFixture.java: Recompile with -Xlint:deprecation for details.
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ oak-segment-azure ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.jackrabbit.oak.segment.azure.AzureRepositoryLockTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 74.816 s - in org.apache.jackrabbit.oak.segment.azure.AzureRepositoryLockTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.8:report (post-unit-test) @ oak-segment-azure ---
[INFO] Skipping JaCoCo execution because property jacoco.skip is set.
[INFO] 
[INFO] --- maven-bundle-plugin:5.1.8:bundle (default-bundle) @ oak-segment-azure ---
[WARNING] Bundle org.apache.jackrabbit:oak-segment-azure:bundle:1.59-SNAPSHOT : Export com.microsoft.azure.storage,  has 4,  private references [com.microsoft.azure.storage.analytics, com.microsoft.azure.storage.file, com.microsoft.azure.storage.queue, com.microsoft.azure.storage.table]
[WARNING] Bundle org.apache.jackrabbit:oak-segment-azure:bundle:1.59-SNAPSHOT : Export com.microsoft.azure.storage.core,  has 2,  private references [com.microsoft.azure.storage.queue, com.microsoft.azure.storage.table]
[WARNING] Bundle org.apache.jackrabbit:oak-segment-azure:bundle:1.59-SNAPSHOT : Export com.microsoft.azure.storage.blob,  has 2,  private references [com.microsoft.azure.keyvault.core, com.microsoft.azure.storage.file]
[INFO] Building bundle: /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/oak-segment-azure-1.59-SNAPSHOT.jar
[INFO] Writing manifest: /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/classes/META-INF/MANIFEST.MF
[INFO] 
[INFO] --- maven-site-plugin:3.12.1:attach-descriptor (attach-descriptor) @ oak-segment-azure ---
[INFO] Skipping because packaging 'bundle' is not pom.
[INFO] 
[INFO] --- maven-jar-plugin:3.3.0:test-jar (default) @ oak-segment-azure ---
[INFO] Building jar: /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/oak-segment-azure-1.59-SNAPSHOT-tests.jar
[INFO] 
[INFO] --- maven-bundle-plugin:5.1.8:baseline (baseline) @ oak-segment-azure ---
[WARNING] com.microsoft.azure.storage: Excessive version increase; detected 1.59.0, suggested 1.58.0
[WARNING] com.microsoft.azure.storage: Version has been increased but analysis detected no changes; detected 1.59.0, suggested 1.58.0
[WARNING] com.microsoft.azure.storage.blob: Excessive version increase; detected 1.59.0, suggested 1.58.0
[WARNING] com.microsoft.azure.storage.blob: Version has been increased but analysis detected no changes; detected 1.59.0, suggested 1.58.0
[WARNING] com.microsoft.azure.storage.core: Excessive version increase; detected 1.59.0, suggested 1.58.0
[WARNING] com.microsoft.azure.storage.core: Version has been increased but analysis detected no changes; detected 1.59.0, suggested 1.58.0
[WARNING] org.apache.jackrabbit.oak.segment.azure: Excessive version increase; detected 1.59.0, suggested 1.58.0
[WARNING] org.apache.jackrabbit.oak.segment.azure: Version has been increased but analysis detected no changes; detected 1.59.0, suggested 1.58.0
[WARNING] org.apache.jackrabbit.oak.segment.azure.util: Excessive version increase; detected 1.59.0, suggested 1.58.0
[WARNING] org.apache.jackrabbit.oak.segment.azure.util: Version has been increased but analysis detected no changes; detected 1.59.0, suggested 1.58.0
[INFO] Baseline analysis complete, 0 error(s), 10 warning(s)
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.8:check (default-check) @ oak-segment-azure ---
[INFO] Skipping JaCoCo execution due to missing execution data file:/Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/coverage-reports/jacoco-ut.exec
[INFO] 
[INFO] --- maven-install-plugin:3.1.0:install (default-install) @ oak-segment-azure ---
[INFO] Installing /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/pom.xml to /Users/trana/.m2/repository/org/apache/jackrabbit/oak-segment-azure/1.59-SNAPSHOT/oak-segment-azure-1.59-SNAPSHOT.pom
[INFO] Installing /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/oak-segment-azure-1.59-SNAPSHOT.jar to /Users/trana/.m2/repository/org/apache/jackrabbit/oak-segment-azure/1.59-SNAPSHOT/oak-segment-azure-1.59-SNAPSHOT.jar
[INFO] Installing /Users/trana/Desktop/PROJECTS/jackrabbit-oak/oak-segment-azure/target/oak-segment-azure-1.59-SNAPSHOT-tests.jar to /Users/trana/.m2/repository/org/apache/jackrabbit/oak-segment-azure/1.59-SNAPSHOT/oak-segment-azure-1.59-SNAPSHOT-tests.jar
[INFO] 
[INFO] --- maven-bundle-plugin:5.1.8:install (default-install) @ oak-segment-azure ---
[INFO] Local OBR update disabled (enable with -DobrRepository)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:19 min
[INFO] Finished at: 2023-10-20T18:32:59+05:30
[INFO] ------------------------------------------------------------------------



Did you by any chance remove or changed this file? 

https://github.com/apache/jackrabbit-oak/blob/trunk/oak-segment-azure/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker

File: oak-blob-cloud-azure/pom.xml (hunk 1 above)
Comment (t-rana) [reply oak-blob-cloud-azure/pom.xml]: Yes we do that, but to be sure that other modules inherit the same version of dependency that is declared in parent module and does not transitively use some other version of same dependency, it will always beneficial to define the version in child modules to avoid dependency conflicts.

If test container dependency is transitively inherited from other dependency, we can end up using some other version than the one defined in parent pom

https://stackoverflow.com/questions/13074231/override-library-version-defined-in-parent-pom

Comment (reschke) [reply oak-blob-cloud-azure/pom.xml]: I still don't get it.

If you believe that we have a general problem with how we maintain version numbers of dependencies in oak-parent, please open a separate ticket and explain the issue - for visibility and so that we can fix the problem systematically.

Otherwise, please just stick with the project conventions.

Comment (t-rana) [reply oak-blob-cloud-azure/pom.xml]: following the project conventions, removed test container dependency version from child modules.

**Review by reschke — COMMENTED** [review]
Diffs now look better to me, thanks.

Will run local tests later.

Would be good to get another review.

**Review by fabriziofortino — APPROVED** [review]
